### PR TITLE
Updates `theme_json_*` filter names with ending `*_data`

### PR DESCRIPTION
**Pull request by kebbet**
Adding `_data` to filter names.
https://core.trac.wordpress.org/ticket/56796

### CODE REVIEW

File: src/wp-includes/class-wp-theme-json-resolver.php [hunk 1]
```diff
@@ -180,7 +180,7 @@ public static function get_core_data() {
 		 *
 		 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
 		 */
-		$theme_json   = apply_filters( 'theme_json_default', new WP_Theme_JSON_Data( $config, 'default' ) );
+		$theme_json   = apply_filters( 'theme_json_default_data', new WP_Theme_JSON_Data( $config, 'default' ) );
```
Comment (ocean90) [inline src/wp-includes/class-wp-theme-json-resolver.php]: Another idea might be to follow a `theme_json_data_{$context}` syntax:

```suggestion
		$theme_json   = apply_filters( 'theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
```

File: src/wp-includes/class-wp-theme-json-resolver.php [hunk 2]
```diff
@@ -490,7 +490,7 @@ public static function get_user_data() {
 				 *
 				 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
 				 */
-				$theme_json = apply_filters( 'theme_json_user', new WP_Theme_JSON_Data( $config, 'custom' ) );
+				$theme_json = apply_filters( 'theme_json_user_data', new WP_Theme_JSON_Data( $config, 'custom' ) );
```
Comment (ocean90) [inline src/wp-includes/class-wp-theme-json-resolver.php]: A bit unrelated, but it's odd that `custom` is used here which isn't documented in https://github.com/wordpress/wordpress-develop/blob/711b74c21bef856677fd97d82d17f7761d0d602a/src/wp-includes/class-wp-theme-json-data.php#L40.

**Comment by SergeyBiryukov** [comment]
Should we consider putting `data` in the middle, e.g. `theme_json_data_default`, etc. to closer match `WP_Theme_JSON_Data`?

Not sure about adding the `wp_` prefix, but seems to make sense here too? `theme_json_get_style_nodes` would then also need the prefix.

**Review by SergeyBiryukov — APPROVED** [review]

**Comment by hellofromtonya** [comment]
Test failures are unrelated to this PR and are currently present in `trunk`.

**Review by hellofromtonya — APPROVED** [review]
LGTM 👍 Ready for commit

**Comment by hellofromtonya** [comment]
Committed via changeset https://core.trac.wordpress.org/changeset/54501. Thank you everyone for your contributions 🎉

**Comment by oandregal** [comment]
This PR needs a few follow-ups:

- [x] [The devnote for 6.1](https://make.wordpress.org/core/2022/10/10/filters-for-theme-json-data/) (it had already been published when this PR was proposed). I've gone ahead and update it.
- [x] The documentation for this in the handbook.
- [x] The same code in Gutenberg.

I've prepared https://github.com/WordPress/gutenberg/pull/44940 to address the last two items. Would anyone be available for review?

File: src/wp-includes/class-wp-theme-json.php [hunk 3]
```diff
@@ -1899,7 +1899,7 @@ protected static function get_style_nodes( $theme_json, $selectors = array() ) {
 		 *
 		 * @param array $nodes Style nodes with metadata.
 		 */
-		return apply_filters( 'theme_json_get_style_nodes', $nodes );
```
Comment (oandregal) [inline src/wp-includes/class-wp-theme-json.php:1902]: @scruffian @ajlende @draganescu I see that, in Gutenberg land, we name this filter `gutenberg_theme_json_style_nodes`. Do we need to take any action here? I haven't seen that we prefix the filters depending on environment (`wp_` or `gutenberg_`) in any other case. My understanding is that it's problematic for any consumer that want to use the filter. Though this rationale may have changed, so I've excluded this from https://github.com/WordPress/gutenberg/pull/44940 and can be looked at separately, if necessary.

Comment (ajlende) [reply src/wp-includes/class-wp-theme-json.php:1902]: Thanks for flagging this @oandregal. I don't see any specific reason for using a different filter in the plugin vs core. I'll open a PR to match this.

EDIT: Opened https://github.com/WordPress/gutenberg/pull/44949

**Pull request closed without merging**